pyomo/solvers/plugins/solvers/CPLEX.py

```diff
@@ -268,14 +275,14 @@ def _convert_priorities_to_rows(self, instance, priorities, directions):
 
             var_direction = directions.get(var, BranchDirection.default)
 
-            if not var.is_indexed():
```

Comment:
I *think* the easiest general solution is:
```python
if hasattr(var, 'values'):
    # catches Var, IndexedVar, and kernel's variable_dict
    var_iter = var.values()
elif hasattr(var, '__iter__'):
    # catches list and kernel's variable_list and variable_tuple
    var_iter = iter(var)
else:
    # kernel's variable
    var_iter = iter((var,))
for v in var_iter:
    if id(v) not in byObject:
        continue
    v_direction = directions.get(v, var_direction)
    rows.append((byObject[id(v)], priority, v_direction))
```

Reply:
Thanks @jsiirola! I've added this logic as a general-purpose helper in `pyomo.util.components` however let me know if you'd rather this logic is kept private to this method or this isn't the best place for it. (To be honest, I'm surprised this hasn't come up before!)